pandas/_libs/tslibs/timezones.pyx

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# cython: profile=False
+
+try: string_types = basestring
+except NameError: string_types = str
+
+
+cdef extern from "Python.h":
+    Py_ssize_t PY_SSIZE_T_MAX
+
+cdef extern from "datetime.h":
+    void PyDateTime_IMPORT()
+
+# import datetime C API
+PyDateTime_IMPORT
+
+
+import numpy as np
+cimport numpy as np
+from numpy cimport ndarray, int64_t, float64_t
+np.import_array()
+
+
+# dateutil compat
+from dateutil.tz import (tzoffset,
+                         tzlocal as _dateutil_tzlocal,
+                         tzfile as _dateutil_tzfile,
+                         tzutc as _dateutil_tzutc,
+                         tzstr as _dateutil_tzstr)
+
+import sys
+if sys.platform == 'win32' or sys.platform == 'cygwin':
```

Comment:
That was copied verbatim from pd.compat, so I'm not sure what idiom it misses.  But as above the idea was to avoid a non-cython import.  Will change.